### PR TITLE
Issue 6861: fill repoIdentifier only for restic repo

### LINKED ISSUE #6861: podvolumebackups still reference Restic in repository ID path (repoIdentifier) after switching to Kopia
**What steps did you take and what happened:**
I recently switched to the Kopia uploader-type and I noticed that my podvolumebackups still show references to Restic in the repository ID path (repoIdentifier).

**What did you expect to happen:**
I did not expect any references to Restic to appear anymore. I had removed all Restic CRDs and the Restic backuprepository as well. Can someone explain to me why Restic is still showing up in the repository ID path?

**The following information will help us better understand what's going on**:
```
NAMESPACE   NAME                STATUS      CREATED   NAMESPACE    POD             VOLUME     REPOSITORY ID                                       UPLOADER TYPE   STORAGE LOCATION   AGE
velero      myschedule          InProgress   4m       redacted     redacted        redacted  s3:https://redacted/velero/redacted/restic/redacted   kopia          default            4m1s
```

**Anything else you would like to add:**

Please let me know if any important information is missing here.

**Environment:**

- Velero version (use `velero version`): `v1.11.1`, installed using Helm Release `4.1.4`
- Velero features (use `velero client config get features`):  not set
- Kubernetes version (use `kubectl version`): v1.25.12
- Kubernetes installer & version: 
- Cloud provider or hardware configuration: on premise
- OS (e.g. from `/etc/os-release`): VMware Tanzu

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Fix #6861. Only Restic path requires repoIdentifier, so for non-restic path, set the repoIdentifier fields as empty in PVB and PVR and also remove the RepoIdentifier column in the get output of PVBs and PVRs

### CODE REVIEW

**Comment by Lyndon-Li** [comment]
@sseago @shubham-pampattiwar 
Please help to review this PR. We want to remove the `Repository ID` column from the output of `kubectl get podvo`lumebackup, since kopia is not the default path and it is always empty for kopia path.

**Comment by reasonerjt** [comment]
Let's double check with other maintainers and adopters about the break change to remove the repo ID when restic is chosen.

File: pkg/apis/velero/v1/pod_volume_backup_types.go [hunk 1]
```diff
@@ -114,7 +114,6 @@ type PodVolumeBackupStatus struct {
 // +kubebuilder:printcolumn:name="Namespace",type="string",JSONPath=".spec.pod.namespace",description="Namespace of the pod containing the volume to be backed up"
 // +kubebuilder:printcolumn:name="Pod",type="string",JSONPath=".spec.pod.name",description="Name of the pod containing the volume to be backed up"
 // +kubebuilder:printcolumn:name="Volume",type="string",JSONPath=".spec.volume",description="Name of the volume to be backed up"
-// +kubebuilder:printcolumn:name="Repository ID",type="string",JSONPath=".spec.repoIdentifier",description="Backup repository identifier for this backup"
```
Comment (shubham-pampattiwar) [inline pkg/apis/velero/v1/pod_volume_backup_types.go:117]: Shouldn't we just remove the `printcolumn` kubebuilder marker ? And keep the repo ID in the CRD, this way we don't display it in the kubectl get call without actually modifying the PVB CRD.

**Review by shubham-pampattiwar — COMMENTED** [review]
@Lyndon-Li I think only the removal of `printcolumn` kubebuilder marker should be sufficient.

File: pkg/apis/velero/v1/pod_volume_backup_types.go (hunk 1 above)
Comment (Lyndon-Li) [reply pkg/apis/velero/v1/pod_volume_backup_types.go:117]: Yes, we didn't change the CRD spec, only removed the marker.

Comment (shubham-pampattiwar) [reply pkg/apis/velero/v1/pod_volume_backup_types.go:117]: ok ok

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**